File: frontend/src/components/enums/status.ts
export enum Status {
	Unknown = 0,
	Loading = 1,
	Ready = 2,
	Streaming = 3,
	Generating = 4,
	Playing = 5,
	Error = 6,
	Warning = 7,
}

export const StatusDisplayNames = {
	[Status.Unknown]: "⚪ Unknown",
	[Status.Loading]: "<div class=\"icon\">&nbsp;</div>️ Loading",
	[Status.Ready]: "🟢 Ready",
	[Status.Streaming]: "🟠 Streaming",
	[Status.Generating]: "🔵 Generating",
	[Status.Playing]: "▶️ Playing",
	[Status.Error]: "🔴 Error",
	[Status.Warning]: "⚠️ Warning",
};

File: app/common/status/status.go
package status

import (
	"nstudio/app/common/eventManager"
)

type ID int

const (
	Unknown ID = iota
	Loading
	Ready
	Streaming
	Generating
	Playing
	Error
	Warning
)

type StatusEvent struct {
	Status  ID     `json:"status"`
	Message string `json:"message"`
}

var currentStatus = StatusEvent{
	Status:  Unknown,
	Message: "",
}

func Set(status ID, message string) {
	currentStatus = StatusEvent{
		Status:  status,
		Message: message,
	}
	eventManager.GetInstance().EmitEvent("status", currentStatus)
}

func Get() StatusEvent {
	return currentStatus
}
